prog/storage/setup_spdk.rb

```diff
@@ -34,19 +34,27 @@ def self.assemble(vm_host_id, version, start_service: false, allocation_weight:
     SpdkInstallation.create(
       version: frame["version"],
       allocation_weight: 0,
-      vm_host_id: vm_host.id
+      vm_host_id: vm_host.id,
+      cpu_count: spdk_cpu_count(total_host_cpus: vm_host.total_cpus)
     ) { _1.id = SpdkInstallation.generate_uuid }
 
     hop_install_spdk
   end
 
   label def install_spdk
-    version = frame["version"]
-    sshable.cmd("sudo host/bin/setup-spdk install #{version.shellescape}")
+    q_version = frame["version"].shellescape
+    cpu_count = spdk_cpu_count(total_host_cpus: vm_host.total_cpus)
+    sshable.cmd("sudo host/bin/setup-spdk install #{q_version} #{cpu_count}")
 
     hop_start_service
   end
 
+  def spdk_cpu_count(total_host_cpus:)
+    # Use 2 SPDK cpus per each 64 host cpus. We round up host cpus to the next
+    # multiple of 64.
+    ((total_host_cpus + 63) / 64) * 2
```

Comment:
Should we cast it to integer?

Reply:
ax161: ((64 + 63)/64) * 2 = 3,96875
ax162: ((96 + 63)/64) * 2 = 4,96875
rx220: ((80 + 63)/64) * 2 = 4,46875

I didn't get how this calculation should work.

Reply:
Roughly well enough ;). Do you remember how I used logarithmic fits for buffer sizing?

The main points we have to get very right are at 32, 48, 80 (arm64) cores. But we also need to get 2-4
cores (development, small) and 64 x64 cores (development, larger) "right enough." Punching out a tiny discontinuous special case for exactly the AX162 seemed a bit weird.

I expect we'll have to make this more complicated over time as the inventory anyone runs clover against expands. It will probably need to start knowing something about architecture and viable hugepage allocations…or, possibly, the way we allocate and tune the number of spdk threads may change.

Reply:
This is to get `ceiling(a/b)` without having to deal with floating point numbers, which are easy to get wrong. If we use integer calculations, `ceiling(a/b) = (a + b - 1) / b`.

Reply:
I rewrote this in a different way, assuming `(a + b - 1) / b` is not a common enough idiom in our circles. But I still assume readers realize this is integer division.

Reply:
I rewrote this again making this range based for now.